packages/client-sdk-nodejs/src/internal/cache-data-client.ts

```diff
@@ -720,6 +736,165 @@ export class CacheDataClient implements IDataClient {
     });
   }
 
+  public async getBatch(
+    cacheName: string,
+    keys: Array<string | Uint8Array>
+  ): Promise<GetBatch.Response> {
+    try {
+      validateCacheName(cacheName);
+    } catch (err) {
+      return this.cacheServiceErrorMapper.returnOrThrowError(
+        err as Error,
+        err => new GetBatch.Error(err)
+      );
+    }
+    this.logger.trace(`Issuing 'getBatch' request; keys: ${keys.toString()}`);
+    const result = await this.sendGetBatch(
+      cacheName,
+      keys.map(key => this.convert(key))
+    );
+    this.logger.trace(`'getBatch' request result: ${result.toString()}`);
+    return result;
+  }
+
+  private async sendGetBatch(
+    cacheName: string,
+    keys: Uint8Array[]
+  ): Promise<GetBatch.Response> {
+    const getRequests = [];
+    for (const k of keys) {
+      const getRequest = new grpcCache._GetRequest({
+        cache_key: k,
+      });
+      getRequests.push(getRequest);
+    }
+    const request = new grpcCache._GetBatchRequest({
+      items: getRequests,
+    });
+    const metadata = this.createMetadata(cacheName);
+
+    const call = this.clientWrapper.getClient().GetBatch(request, metadata, {
+      interceptors: this.streamingInterceptors,
+    });
+
+    return await new Promise((resolve, reject) => {
+      const results: CacheGet.Response[] = [];
+      call.on('data', (getResponse: grpcCache._GetResponse) => {
+        const result = getResponse.result;
+        switch (result) {
+          case grpcCache.ECacheResult.Hit:
+            results.push(new CacheGet.Hit(getResponse.cache_body));
+            break;
+          case grpcCache.ECacheResult.Miss:
+            results.push(new CacheGet.Miss());
+            break;
+          default:
+            results.push(
+              new CacheGet.Error(new UnknownError(getResponse.message))
+            );
+        }
+      });
+
+      call.on('end', () => {
+        resolve(new GetBatch.Success(results, keys));
+      });
+
+      call.on('error', (err: ServiceError | null) => {
+        this.cacheServiceErrorMapper.resolveOrRejectError({
+          err: err,
+          errorResponseFactoryFn: e => new GetBatch.Error(e),
+          resolveFn: resolve,
+          rejectFn: reject,
+        });
+      });
```

Comment:
For the backlog: (1) the three callbacks -- data, end and error -- have boilerplate we can abstract away; and (2) we should require the developer to implement all three.

Otherwise 👍

Reply:
@malandis wdym? is your comment targeted at future streaming APIs?

Reply:
or just consolidating the get/set ones I guess?